c7n/resources/cw.py

```diff
@@ -705,6 +705,59 @@ def process_resource_set(self, client, accounts, resources):
         return results
 
 
+@LogGroup.filter_registry.register('subscription-filter')
+class LogSubscriptionFilter(ValueFilter):
+    """Filters CloudWatch log groups by subscriptions
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: cloudwatch-groups-with-subscriptions
+                resource: log-group
+                filters:
+                  - type: subscription-filter
+                    key: destinationArn
+                    value: arn:aws:lambda:us-east-1:123456789876:function:forwarder
+    """
+    schema = type_schema('subscription-filter', rinherit=ValueFilter.schema)
+    annotation_key = 'c7n:SubscriptionFilters'
+    permissions = ('logs:DescribeSubscriptionFilters',)
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('logs')
+        results = []
+        with self.executor_factory(max_workers=1) as w:
```

Comment:
given max workers 1 we can just walk through the set without the executor factory.

Reply:
OK. I stole that from from cross-account filter. Would you like me to fix that one too?

Reply:
Resolved by 8d45c5ba403c8e72ee8c98cacc63d250e5578bb6